Components/VxWatchDogTimer/VxWatchDogTimer.cpp

```diff
@@ -0,0 +1,70 @@
+// ======================================================================
+// \title  VxWatchDogTimer.cpp
+// \author ortega
+// \brief  cpp file for VxWatchDogTimer component implementation class
+// ======================================================================
+
+#include "Components/VxWatchDogTimer/VxWatchDogTimer.hpp"
+#include <sysLib.h>
+
+namespace Components {
+
+// ----------------------------------------------------------------------
+// Component construction and destruction
+// ----------------------------------------------------------------------
+
+VxWatchDogTimer ::VxWatchDogTimer(const char* const compName) : VxWatchDogTimerComponentBase(compName) {
+    this->m_watchdogId = wdCreate();
+}
+
+VxWatchDogTimer ::~VxWatchDogTimer() {
+    if (this->m_watchdogId != nullptr) {
+        (void)wdCancel(this->m_watchdogId);
+        (void)wdDelete(this->m_watchdogId);
+    }
+}
+
+void VxWatchDogTimer::startWatchdog(U32 ticks) {
+    this->m_tickDelay = static_cast<_Vx_ticks_t>(ticks);
+    FW_ASSERT(this->m_watchdogId != nullptr);
+    STATUS status = wdStart(this->m_watchdogId, this->m_tickDelay, reinterpret_cast<FUNCPTR>(s_driverISR),
+                            reinterpret_cast<_Vx_usr_arg_t>(this));
+    FW_ASSERT(status == VXWORKS_OK);
+}
+
+void VxWatchDogTimer::startWatchdog(Fw::TimeInterval interval) {
+    static constexpr U32 MS_PER_SECS = 1000;
+    static constexpr U32 USECS_PER_MS = 1000;
+    U64 delayInMs = (interval.getSeconds() * MS_PER_SECS) + (interval.getUSeconds() / USECS_PER_MS);
```

Comment:
I am ok with this, but it does require F´ and 64bit enabled.  There are ways to check overflow without it.  For example, you could do:

```
U32 delaySubseconds = interval.getUSeconds() / USECS_PER_MS; // Well bounded (0-1000)
FW_ASSERT((std::numeric_limits<U32>::max() - delaySubseconds)/ MS_PER_SECS >= interval.getSeconds()); // Check for overflow if seconds would overflow

```